kedro-datasets/kedro_datasets/spark/spark_streaming_dataset.py

```diff
@@ -0,0 +1,200 @@
+"""SparkStreamingDataSet to load and save a PySpark Streaming DataFrame."""
+import json
+from copy import deepcopy
+from pathlib import PurePosixPath
+from typing import Any, Dict
+
+import fsspec
+from kedro.io.core import (
+    AbstractDataSet,
+    DataSetError,
+    get_filepath_str,
+    get_protocol_and_path,
+)
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.types import StructType
+from pyspark.sql.utils import AnalysisException
+
+from kedro_datasets.spark.spark_dataset import _split_filepath, _strip_dbfs_prefix
+
+
+class SparkStreamingDataSet(AbstractDataSet):
+    """``SparkStreamingDataSet`` loads data into Spark Streaming Dataframe objects.
+    Example usage for the
+    `YAML API <https://kedro.readthedocs.io/en/stable/data/\
+    data_catalog.html#use-the-data-catalog-with-the-yaml-api>`_:
+    .. code-block:: yaml
+        raw.new_inventory:
+            type: spark.SparkStreamingDataSet
+            filepath: data/01_raw/stream/inventory/
+            file_format: json
+        int.new_inventory:
+            type: spark.SparkStreamingDataSet
+            filepath: data/02_intermediate/inventory/
+            file_format: csv
+            save_args:
+                output_mode: append
+                checkpoint: data/04_checkpoint/int_new_inventory
+                header: True
+            load_args:
+                header: True
```

Comment:
Does this work? Do you need the `schema` specified? I would consider making schema a separate argument, since it is necessary for both save/load instead of just `load_args` now.

Reply:
Yes the schema doc is updated.